### PR TITLE
Fix API of DoLaterTimer to match earlier versions

### LINKED ISSUE #388: Change to API of `DoLaterTimer`
The most recent release of PyFace changed the `DoLaterTimer` API: with the previous API, a timer was instantiated using:
```
my_timer = DoLaterTimer(time_interval, callback, args, kwargs)
```
Where `args` and `kwargs` are a tuple and dictionary of positional and named arguments, as usual.

But the current API requires instantiation via:
```
my_timer = DoLaterTimer(time_interval, callback, *args, **kwargs)
```

### PR DESCRIPTION
**Pull request by corranwebster**
Fixes #388.  With a regression test, even.

### CODE REVIEW

File: pyface/timer/tests/test_do_later.py [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+        finally:
+            timer.stop()
+
+        self.assertEqual(handler.count, 1)
+
+        expected_time = start_time + 0.1
+
+        # give feedback in case of failure
+        if expected_time > handler.times[0]:
+            print(expected_time, handler.times)
```
Comment (mdickinson) [inline pyface/timer/tests/test_do_later.py]: Can we put this information into the failure message instead of `print`ing it? (Same comment again below.)

**Comment by mdickinson** [comment]
Looks like the test is fragile:
```
======================================================================
FAIL: test_basic (pyface.timer.tests.test_do_later.TestDoAfter)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/.edm/envs/pyface-test-3.6-pyside2/lib/python3.6/site-packages/pyface-6.2.0.dev15-py3.6.egg/pyface/timer/tests/test_do_later.py", line 117, in test_basic
    self.assertTrue(expected_time <= handler.times[0])
AssertionError: False is not true
-------------------- >> begin captured stdout << ---------------------
337.10411175400003 [337.100504496]
--------------------- >> end captured stdout << ----------------------
```

File: pyface/timer/tests/test_do_later.py (hunk 1 above)
Comment (corranwebster) [reply pyface/timer/tests/test_do_later.py]: Did this. Apparently I can still learn things about the Python standard library, I'd never used the feature of adding a failure message to a unittest assert method.

**Comment by corranwebster** [comment]
I've added a little slack to the test - as long as it takes more than 95 milliseconds we'll consider it good enough, since these aren't meant for precision timing (in particular they can take much longer).

**Pull request merged**